p2p/ipld/sample.go

```diff
@@ -0,0 +1,102 @@
+package ipld
+
+import (
+	"github.com/ipfs/go-cid"
+	"github.com/lazyledger/nmt/namespace"
+
+	"github.com/lazyledger/lazyledger-core/libs/rand"
+	"github.com/lazyledger/lazyledger-core/p2p/ipld/plugin/nodes"
+	"github.com/lazyledger/lazyledger-core/types"
+)
+
+// Sample is a point in 2D space over square.
+type Sample struct {
+	Row, Col uint32
+}
+
+// SampleSquare randomly picks *num* unique points from arbitrary *width* square
+// and returns them as samples.
+func SampleSquare(squareWidth uint32, num int) []Sample {
+	ss := newSquareSampler(squareWidth, num)
+	ss.sample(num)
+	return ss.sampled()
```

Comment:
Shouldn't this be named `ss.samples()` instead of `ss.sampled()`?

Reply:
> Hey fellow squareSampler, pls return me what you've sampled! 
Also, I named `sampled` not to cross naming with internal map, though from another view I think `samples` is better and changed internal map naming to C-style - `smpls`